src/site/_filters/patterns.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Finds a pattern by the pattern's id in the paaterns data object.
+ * @param {string} id Pattern Id
+ * @param {Array} patterns Patterns data object
+ * @return {Object} Pattern object
+ */
+const getPattern = (id, patterns) => {
+  return patterns.find((item) => item.id === id);
```

Comment:
I'm super nitpicking but this method implies that it should be stored in a dictionary, not a list. But we have O(1) things so ¯‍\‍_‍(‍ツ‍)‍_‍/‍¯

Reply:
We want patterns to resemble eleventy collections, so it has to behave like a list. I think replicating it as a dictionary for quick access by key is not critical here.

Reply:
Ignoring if it's critical or not, I don't know if having a collection as an array is that important.

1) I think 11ty does it only because the built in collections are just filtering all content based on tags, so that makes sense in a way.
2) 11ty pagination accepts objects also, not just arrays.
3) Collections such as shows, authors, and tags are objects (key/values), so we already have dict like collections.

I guess I'm just saying the whole array collection isn't really necessary and I don't think we need to resemble 11ty's built in collections.

Reply:
Ok, let's try to have it as an object.